worlds/messenger/SubClasses.py

```diff
@@ -57,11 +57,11 @@ def __init__(self, name: str, parent: MessengerRegion, loc_id: Optional[int]) ->
 class MessengerShopLocation(MessengerLocation):
     def cost(self) -> int:
         name = self.name.replace("The Shop - ", "")  # TODO use `remove_prefix` when 3.8 finally gets dropped
-        world = self.parent_region.multiworld.worlds[self.player]
+        world: MessengerWorld = self.parent_region.multiworld.worlds[self.player]
```

Comment:
Not blocking the PR with this, but you would have to use `typing.cast()` here to get mypy to accept your code.

There are a couple of other issues in this file. Some of them are because core is not ideal for typed python, some would strictly require a bunch of asserts (to tell mypy that an Optional[Something] is expected to not be None).